lineman/app/components/change_password_form/change_password_form.coffee

```diff
@@ -0,0 +1,14 @@
+angular.module('loomioApp').factory 'ChangePasswordForm', ->
+  templateUrl: 'generated/components/change_password_form/change_password_form.html'
+  controller: ($scope, $rootScope, CurrentUser, Records, FlashService) ->
+    $scope.user = CurrentUser.clone()
+
+    $scope.submit = ->
+      $scope.isDisabled = true
+      Records.users.changePassword($scope.user).then ->
+        $scope.isDisabled = false
+        FlashService.success('profile_page.messages.password_changed')
+        $scope.$close()
+      , ->
+        $scope.isDisabled = false
+        # $rootScope.$broadcast 'pageError', 'cantChangePassword', $scope.user
```

Comment:
It's coming up on a time where I'd start to feel confident about writing a generalized version of how to handle validation errors, server errors, and successes on forms in a general way. Maybe not quite yet, but SOOON